client/tinychain/ml/linalg.py

```diff
@@ -283,29 +283,27 @@ def householder_bidiag(U: Tensor, W: Tensor, e: Tensor):
     """Householder's reduction to bidiagonal form"""
     @post_op
     def check_rows_num(cxt, U: Tensor, scale: F32, i: UInt, m: UInt, n: UInt, l: UInt):
-        scale = F32(tc.linalg.norm(U[i:, i]))**2
+        scale = F32(norm(U[i:, i]))**2
 
         # @post_op
         def change_rows(cxt, U: Tensor, scale: F32, i: UInt, l: UInt, n: UInt):
             s = F32(After(U[i:, i].write(U[i:, i] / scale),
-                    F32(tc.linalg.norm(U[i:, i]))))
-            f = F32(U[i, i])
-            g = F32(If(f >= F32(0), -s, s))*1
-            h = F32(f * g - s)
-            U[i, i].write(f - g)
+                    F32(norm(U[i:, i]))))
```

Comment:
nit: this `F32` is redundant with the one outside the `After`

PTAL at https://github.com/haydnv/tinychain/pull/68 because I think it could remove a lot of these repetitive type declarations